gcloud-java-compute/src/main/java/com/google/gcloud/spi/DefaultComputeRpc.java

```diff
@@ -43,4 +64,214 @@ public DefaultComputeRpc(ComputeOptions options) {
   private static ComputeException translate(IOException exception) {
     return new ComputeException(exception);
   }
+
+  @Override
+  public DiskType getDiskType(String zone, String diskType, Map<Option, ?> options)
+      throws ComputeException {
+    try {
+      return compute.diskTypes()
+          .get(this.options.projectId(), zone, diskType)
+          .setFields(FIELDS.getString(options))
+          .execute();
+    } catch (IOException ex) {
+      ComputeException serviceException = translate(ex);
+      if (serviceException.code() == HTTP_NOT_FOUND) {
+        return null;
+      }
+      throw serviceException;
+    }
+  }
+
+  @Override
+  public Tuple<String, Iterable<DiskType>> listDiskTypes(String zone, Map<Option, ?> options)
+      throws ComputeException {
+    try {
+      DiskTypeList diskTypesList = compute.diskTypes()
+          .list(this.options.projectId(), zone)
+          .setFilter(FILTER.getString(options))
+          .setMaxResults(MAX_RESULTS.getLong(options))
+          .setPageToken(PAGE_TOKEN.getString(options))
+          .setFields(FIELDS.getString(options))
+          .execute();
+      Iterable<DiskType> diskTypes = diskTypesList.getItems();
+      return Tuple.of(diskTypesList.getNextPageToken(), diskTypes);
+    } catch (IOException ex) {
+      throw translate(ex);
+    }
+  }
+
+  @Override
+  public Tuple<String, Iterable<DiskType>> listDiskTypes(Map<Option, ?> options)
+      throws ComputeException {
+    try {
+      DiskTypeAggregatedList aggregatedList = compute.diskTypes()
+          .aggregatedList(this.options.projectId())
+          .setFilter(FILTER.getString(options))
+          .setMaxResults(MAX_RESULTS.getLong(options))
+          .setPageToken(PAGE_TOKEN.getString(options))
+          .setFields(FIELDS.getString(options))
+          .execute();
+      ImmutableList.Builder<DiskType> builder = ImmutableList.builder();
+      Map<String, DiskTypesScopedList> scopedList = aggregatedList.getItems();
+      if (scopedList != null) {
+        for (String key : scopedList.keySet()) {
+          DiskTypesScopedList diskTypesScopedList = scopedList.get(key);
+          if (diskTypesScopedList.getDiskTypes() != null) {
+            builder.addAll(diskTypesScopedList.getDiskTypes());
+          }
+        }
+      }
+      return Tuple.<String, Iterable<DiskType>>of(aggregatedList.getNextPageToken(),
+          builder.build());
+    } catch (IOException ex) {
+      throw translate(ex);
+    }
+  }
+
+  @Override
+  public MachineType getMachineType(String zone, String machineType, Map<Option, ?> options)
+      throws ComputeException {
+    try {
+      return compute.machineTypes()
+          .get(this.options.projectId(), zone, machineType)
+          .setFields(FIELDS.getString(options))
+          .execute();
+    } catch (IOException ex) {
+      ComputeException serviceException = translate(ex);
```

Comment:
if you wanted you can make this re-occuring code a generic method that returns T (always null) or throws ComputeException.